docs/release_notes.adoc

```diff
@@ -5,7 +5,23 @@ include::include.adoc[]
 
 === Breaking Changes
 
+==== Sputnik Runner removed (an alternative)
+
+In 2.0-M1 the Sputnik runner was removed and the 2.0-M2 release notes explicitly mentioned that enhancements
+that extend `Sputnik` or use it as a delegate like for example the `PowerMockRunnerDelegate` will not work anymore.
+
+This is not the full truth though. There is a JUnit 4 runner provided by JUnit 5 called `JUnitPlatform` that can
+run any JUnit platform based tests - like Spock 2+ based tests - in a JUnit 4 environment. This is intended
+for situations where an IDE, build tool, CI server, or similar does not yet natively support JUnit platform.
+It is provided in the artifact `org.junit.platform:junit-platform-runner`.
+
+This means, if you make sure your tests are launched with JUnit 4 and you use `JUnitPlatform` where you before used
+`Sputnik`, all should work out properly. In case of PowerMock this means you annotate your specification with
+`@RunWith(PowerMockRunner)` and `@PowerMockRunnerDelegate(JUnitPlatform)` and make sure your tests are launched
+using JUnit 4.
```

Comment:
What is the status of that runner? Is it treated as legacy, fallback approach? If yes, maybe it would be good mention that the native JUnit 5/Spock-based alternatives (if available) are preferred (as e.g. JUnit 6 could have it dropped)?

Reply:
@marcphilipp can you answer the question about status?

Reply:
@szpak I added a short para with some appeal to the conscience. wdyt?

Reply:
Thanks. LGTM. However, I don't have wide experience with runners in JUnit 5, so it would be good to hear @marcphilipp or @leonard84 's opition, who wrote the Spock integration with JUnit Platform.

Reply:
The runner is marked as "stable". However, it was originally meant as a interim solution for builds/environments that did not support running JUnit Platform natively, yet.

Reply:
I think the German expression "von Hinten durch die Brust ins Auge" best describes this roundabout workaround, do we really want to mention it? You can't really upgrade to JUnit 5 as main engine using this, or can this be executed via the `vintage-engine` @marcphilipp?

Reply:
No, you cannot, otherwise you would end up with an endless loop if you use that approach and also have the vintage engine on the class path.
The vintage engine recognizes that runner and refuses to enter a loop.
Actually there is no point in using the vintage engine with that approach.

With the vintage engine, the gist is, that your tests run on JUnit 5 but it can "downgrade" to to JUnit 4, running legacy tests.

With this runner it is the opposite, you run your tests on JUnit 4 but it can "upgrade" to JUnit 5, running platform tests.

You use JUnit 4 as "kickstarter" so to say.
This makes the vintage engine unneeded actually as all tests that do not have the `@RunWith` annotation are run with JUnit 4 anyway.

That phrase was exactly what I had in mind too when I made this tactic up, but the point is it works and without much hassle.

The problem I see is, especially as you added to the release notes that all that depends on Sputnik is not usable anymore, that this can hinder people to migrate to Spock 2+ if they depend on certain things, like PowerMock.

I wouldn't and didn't add this tactic to the normal user guide.
But I thought that it is nice to have them in the release notes, so that people that read the release notes get a hint how to be able to use the new version and still use things like PowerMock without a problem.

They just need to annotate their specifications with `@RunWith(JUnitPlatform)` which can even be done in some base specification and the ones they need for example PowerMock in with `@RunWith(PowerMockRunner) @PowerMockRunnerDelegate(JUnitPlatform)` and run their tests with JUnit 4 and all works properly.

Actually we could even be super-nice and add two AST transformations, one that detects `@RunWith(PowerMockRunner)` and automatically add `@PowerMockRunnerDelegate(JUnitPlatform)` and one that detects the presence of the class `JUnitPlatform` and automatically add `@RunWith(JUnitPlatform)` to all specifications that do not have a `@RunWith` annotation.

Then the only thing left to the user would be to make sure the tests run with JUnit 4 if PowerMock is needed and add `@RunWith(PowerMockRunner)` to the tests where PowerMock is needed. And if no PowerMock is needed, they can simply run with JUnit 5, as there will be no annotations added as `JUnitPlatform` is not present and even if it were, it would just be ignored.

Reply:
Before we add any ASTs I'd say #735 should be done instead. 

I'd suggest to move the line saying this is not recommended to the beginning and use stronger wording to discourage this use/make sure that we don't support this officially in any way.

Reply:
> Before we add any ASTs I'd say #735 should be done instead.

Definitely, go an and do it. :-D ;-)

Actually I'm playing around with ByteBuddy right now to get a hang on it and maybe come up with something, but don't expect anyhting anytime soon, if at all. :-)

I didn't say we should add such AST transformations, I just said we could, to make it easier to be used.
Actually it could even be done as a 3rd-party thing, or added to PowerMock.

But yes, a proper Power-Spock would of course be better.
While just being able to mock final classes and methods - what #735 is about - would not be sufficient to remove the need to use something like PowerMock. Mocking static methods and suppressing static initializers only being two points.

Reply:
I pushed a new version, have a look :-)